src/handler/retrieve-fields.ts

```diff
@@ -0,0 +1,96 @@
+import { RequestHandler } from "express-serve-static-core";
+import { v4 as uuid } from "uuid";
+import { pipe } from "fp-ts/lib/function";
+import { formatValidationErrors } from "io-ts-reporters";
+import * as E from "fp-ts/lib/Either";
+import * as TE from "fp-ts/lib/TaskEither";
+import { logger } from "../logger";
+import { config } from "../config";
+import { WalletFieldsResponse } from "../generated/wallet/WalletFieldsResponse";
+import { Field } from "../generated/wallet/Field";
+import { ProblemJson } from "../generated/wallet/ProblemJson";
+
+const NPG_API_KEY = config.NPG_API_KEY;
+
+export const internalServerError = (): ProblemJson => ({
+  detail: "Internal Server Error",
+  title: "Invalid npg body response"
+});
+
+export const buildWalletFieldsResponse = (
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  sessionResponse: any
+): WalletFieldsResponse => ({
+  cardFormFields: sessionResponse.fields as ReadonlyArray<Field>,
+  sessionId: sessionResponse.sessionId
+});
+
+export const createFormWithNpg: RequestHandler = async (_req, res) => {
+  const walletId = _req.params.walletId;
+  if (walletId == null) {
+    logger.error("Missing walletId param!");
+    return res.status(400).send("Missing walletId param!");
+  }
+
+  logger.info(
+    `[Invoke NPG for create form using walletId: ${_req.params.walletId}] - Return success case`
+  );
+
+  const orderId = uuid().substring(0, 15);
+  const postData = JSON.stringify({
+    merchantUrl: `${_req.protocol}://${_req.get("Host")}`,
+    order: {
+      amount: "0",
+      currency: "EUR",
+      orderId
+    },
+    paymentSession: {
+      actionType: "VERIFY",
+      amount: "0",
+      cancelUrl: "http://payment-wallet.pagopa.it/cancel",
+      captureType: "IMPLICIT",
+      language: "ITA",
+      notificationUrl: "http://payment-wallet.pagopa.it",
+      paymentService: "CARDS",
+      resultUrl: "http://payment-wallet.pagopa.it/esito"
```

Comment:
For a future PR I would make this configurable so that one can bind them to localhost when developing `wallet-fe` 🚀

Reply:
Fix with d7d322651d86304b553d7fdae8bed11825396fab 🚀